parsec/serde/fields.py

```diff
@@ -72,20 +72,19 @@
 class BaseEnumField(Field):
     ENUM: Enum
 
-    def _serialize(self, value, attr, obj):
+    def _serialize(self, value: Any, attr: Any, obj: Any) -> str | None:
         if value is None:
             return None
-
-        if not isinstance(value, self.ENUM):
-            raise ValidationError(f"Not a {self.ENUM.__name__}")
+        if not isinstance(value, tuple([type(x) for x in self.ENUM])):  # type: ignore[attr-defined]
+            raise ValidationError(f"Not a {self.ENUM.__name__}")  # type: ignore[attr-defined]
 
         return value.value
 
-    def _deserialize(self, value, attr, data):
+    def _deserialize(self, value: str, attr: Any, data: Any) -> Enum:
```

Comment:
From Marshmallow `_deserialize` docstring we have:
```
Help on function _deserialize in module marshmallow.fields:

_deserialize(self, value, attr, data)
    Deserialize value. Concrete :class:`Field` classes should implement this method.

    :param value: The value to be deserialized.
    :param str attr: The attribute/key in `data` to be deserialized.
    :param dict data: The raw input data passed to the `Schema.load`.
    :raise ValidationError: In case of formatting or validation failure.
    :return: The deserialized value.
```
I think you can type `attr` to be a `str` instead of `Any`. Stubs for `marshmallow` are also added in #3283 if this can help.

Reply:
Done